libvmi/arch/amd64.c

```diff
@@ -241,6 +241,12 @@ status_t v2p_ia32e (vmi_instance_t vmi,
     return status;
 }
 
+static inline
+bool is_duplicate(vmi_instance_t vmi, uint64_t *page, uint64_t index)
+{
+    return vmi->os_type == VMI_OS_WINDOWS && index > 0 && page[index] == page[index - 1];
```

Comment:
Is it safe to access `page[index - 1]` for index 0?

Reply:
`page[index - 1]` will not be accessed for index 0 due to short-circuit evaluation of the `&&` operator and the previous check against `index > 0`. See [ISO/IEC 9899 standard, section 6.5.13](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf) for this:

> Unlike the bitwise binary & operator, the && operator guarantees left-to-right evaluation;
there is a sequence point after the evaluation of the first operand. If the first operand
compares equal to 0, the second operand is not evaluated.